### ISSUE TITLE
DefaultAzureCredential exception issues

### ISSUE THREAD

**Issue by jongio** [Azure/azure-sdk-for-net#8936]
I get this exception without any creds set.

Overall there is way too much information returned and duplicated in this error message.  

Suggestion: Only display exception messages once and clean them up as indicated below.

`ManagedIdentityCredential is unavailable No managed identity endpoint found..`
 - It is missing a period after unavailable
 - It ends with two periods

`SharedTokenCacheCredential is unavailable No accounts were found in thecache.  To authenticate with the SharedTokenCacheCredential, login an account through developer tooling supporting Azure single sign on..`

 - Same as above. Missing period. Ends with two periods.
 - "thecache" needs a space.
 - "login" should be "log in"
 - Suggest this: "To authenticate with the SharedTokenCacheCredential, log in with a development tool that supports Azure single sign on."

```bash
Unhandled exception. System.AggregateException: One or more errors occurred. (The DefaultAzureCredential failed to retrieve a token from the included credentials.
  EnvironmentCredential is unavailable Environment variables not fully configured. AZURE_TENANT_ID and AZURE_CLIENT_ID must be set, along with either AZURE_CLIENT_SECRET or AZURE_USERNAME and AZURE_PASSWORD. Currently set variables [  ].
  ManagedIdentityCredential is unavailable No managed identity endpoint found..
  SharedTokenCacheCredential is unavailable No accounts were found in thecache.  To authenticate with the SharedTokenCacheCredential, login an account through developer tooling supporting Azure single sign on..
See inner exception for more detail.)
 ---> Azure.Identity.AuthenticationFailedException: The DefaultAzureCredential failed to retrieve a token from the included credentials.
  EnvironmentCredential is unavailable Environment variables not fully configured. AZURE_TENANT_ID and AZURE_CLIENT_ID must be set, along with either AZURE_CLIENT_SECRET or AZURE_USERNAME and AZURE_PASSWORD. Currently set variables [  ].
  ManagedIdentityCredential is unavailable No managed identity endpoint found..
  SharedTokenCacheCredential is unavailable No accounts were found in thecache.  To authenticate with the SharedTokenCacheCredential, login an account through developer tooling supporting Azure single sign on..
See inner exception for more detail.
 ---> System.AggregateException: The DefaultAzureCredential failed to retrieve a token from the included credentials. (Environment variables not fully configured. AZURE_TENANT_ID and AZURE_CLIENT_ID must be set, along with either AZURE_CLIENT_SECRET or AZURE_USERNAME and AZURE_PASSWORD. Currently set variables [  ]) (No managed identity endpoint found.) (No accounts were found in thecache.  To authenticate with the SharedTokenCacheCredential, login an account through developer tooling supporting Azure single sign on.)
 ---> Azure.Identity.CredentialUnavailableException: Environment variables not fully configured. AZURE_TENANT_ID and AZURE_CLIENT_ID must be set, along with either AZURE_CLIENT_SECRET or AZURE_USERNAME and AZURE_PASSWORD. Currently set variables [  ]
   --- End of inner exception stack trace ---
 ---> (Inner Exception #1) Azure.Identity.CredentialUnavailableException: No managed identity endpoint found.<---

 ---> (Inner Exception #2) Azure.Identity.CredentialUnavailableException: No accounts were found in thecache.  To authenticate with the SharedTokenCacheCredential, login an account through developer tooling supporting Azure single sign on.<---

   --- End of inner exception stack trace ---
   at Azure.Identity.DefaultAzureCredential.GetTokenAsync(Boolean isAsync, TokenRequestContext requestContext, CancellationToken cancellationToken)
   at Azure.Identity.DefaultAzureCredential.GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
   at Azure.Core.Pipeline.BearerTokenAuthenticationPolicy.ProcessAsync(HttpMessage message, ReadOnlyMemory`1 pipeline, Boolean async)
   at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.ProcessAsync(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.RetryPolicy.ProcessAsync(HttpMessage message, ReadOnlyMemory`1 pipeline, Boolean async)
   at Azure.Core.Pipeline.RetryPolicy.ProcessAsync(HttpMessage message, ReadOnlyMemory`1 pipeline, Boolean async)
   at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.ProcessAsync(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.ProcessAsync(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Storage.Blobs.BlobRestClient.Container.CreateAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, PublicAccessType access, Nullable`1 timeout, IDictionary`2 metadata, String requestId, Boolean async, String operationName, CancellationToken cancellationToken)
   at Azure.Storage.Blobs.BlobContainerClient.CreateInternal(PublicAccessType publicAccessType, IDictionary`2 metadata, Boolean async, CancellationToken cancellationToken, String operationName)
   at Azure.Storage.Blobs.BlobContainerClient.CreateIfNotExistsInternal(PublicAccessType publicAccessType, IDictionary`2 metadata, Boolean async, CancellationToken cancellationToken)
   at Azure.Storage.Blobs.BlobContainerClient.CreateIfNotExistsAsync(PublicAccessType publicAccessType, IDictionary`2 metadata, CancellationToken cancellationToken)
   at blobmanid.Program.CreateBlockBlobAsync(String accountName, String containerName, String blobName) in C:\Users\jong\OneDriveMS\code\azure-blob-storage-test\blobmanid\Program.cs:line 37
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at blobmanid.Program.Main(String[] args) in C:\Users\jong\OneDriveMS\code\azure-blob-storage-test\blobmanid\Program.cs:line 19
```

Related to #8665

**Comment by jongio** [comment]
@tzhanl - Please do a PR to fix these issues.

**Comment by tzhanl** [comment]
@jongio, PR[https://github.com/Azure/azure-sdk-for-net/pull/9038](https://github.com/Azure/azure-sdk-for-net/pull/9038)

**Comment by jongio** [comment]
More:

Please fix the spacing around these brackets and put a comma in between variable names 

`Currently set variables [  AZURE_TENANT_ID AZURE_CLIENT_SECRET ].`

Also, put in an invalid tenant id and you get this:

Please ensure the fix includes this as well.  For example, we say 'unhandled exception' 4 times in the first sentence.

```
Azure.Identity.AuthenticationFailedException: The DefaultAzureCredential failed due to an unhandled exception:  EnvironmentCredential failed with unhandled exception The authentication request failed due to an unhandled exception.  See inner exception for details..
  EnvironmentCredential failed with The authentication request failed due to an unhandled exception.  See inner exception for details..
See inner exception for more detail.
 ---> System.AggregateException: The DefaultAzureCredential failed due to an unhandled exception:  EnvironmentCredential failed with unhandled exception The authentication request failed due to an unhandled exception.  See inner exception for details.. (The authentication request failed due to an unhandled exception.  See inner exception for details.)
 ---> Azure.Identity.AuthenticationFailedException: The authentication request failed due to an unhandled exception.  See inner exception for details.
 ---> Azure.RequestFailedException:
Status: 400 (Bad Request)

Content:
{"error":"invalid_request","error_description":"AADSTS90002: Tenant '72f988bf-86f1-41af-91ab-2d7cd011db47f' not found. This may happen if there are no active subscriptions for the tenant. Check to make sure you have the correct tenant ID. Check with your subscription administrator.\r\nTrace ID: dfc6e6ae-ed95-41c1-8e04-75186e9a0a00\r\nCorrelation ID: 0d2c3e24-a166-433b-8240-99f2285e5084\r\nTimestamp: 2019-12-09 19:04:31Z","error_codes":[90002],"timestamp":"2019-12-09 19:04:31Z","trace_id":"dfc6e6ae-ed95-41c1-8e04-75186e9a0a00","correlation_id":"0d2c3e24-a166-433b-8240-99f2285e5084","error_uri":"https://login.microsoftonline.com/error?code=90002"}

```

@tzhanl

**Comment by schaabs** [comment]
Fixed by #9821
